appintro/src/main/java/com/github/paolorotolo/appintro/AppIntro2.kt

```diff
@@ -3,32 +3,32 @@ package com.github.paolorotolo.appintro
 import android.graphics.drawable.Drawable
 import android.os.Bundle
 import android.view.View
-import android.widget.FrameLayout
 import android.widget.ImageButton
-import android.widget.LinearLayout
 import androidx.annotation.ColorInt
+import androidx.constraintlayout.widget.ConstraintLayout
 
 abstract class AppIntro2 : AppIntroBase() {
 
-    var backgroundView: View? = null
+    var backgroundView: Int? = null
         set(value) {
             field = value
             if (field != null) {
-                backgroundFrame.addView(field)
+                backgroundFrame.setBackgroundResource(field!!)
```

Comment:
```suggestion
                backgroundFrame.setBackgroundResource(field)
```

Reply:
It gives error when I remove the double exclamation marks: Smart cast to 'Int' is impossible, because 'field' is a mutable property that could have been changed by this time

Reply:
Then everything should be replaced with:
```
field?.let { backgroundFrame.setBackgroundResource(it) }
```